interfaces/builtin/pulseaudio.go

```diff
@@ -23,27 +23,157 @@ import (
 	"github.com/snapcore/snapd/interfaces"
 )
 
-const pulseaudioConnectedPlugAppArmor = `
-/etc/pulse/ r,
-/etc/pulse/* r,
+var pulseaudioConnectedPlugAppArmor = []byte(`
+capability dac_override,
+
+/etc/machine-id r,
+
 /{run,dev}/shm/pulse-shm-* rwk,
-owner /{,var/}run/user/*/pulse/ r,
+
+# FIXME Desktop-only
+owner /{,var/}run/user/*/pulse/ rwk,
 owner /{,var/}run/user/*/pulse/native rwk,
-`
 
-const pulseaudioConnectedPlugSecComp = `
+# Running as system instance
+owner /{,var/}run/pulse rwk,
+owner /{,var/}run/pulse/native rwk,
+`)
+
+var pulseaudioConnectedPlugAppArmorDesktop = []byte(`
+# Only on desktop we need access to /etc/pulse for any PulseAudio client
+# to read available client side configuration settings. On an Ubuntu Core
+# device those things will be stored inside the snap directory.
+/etc/pulse/ r,
+/etc/pulse/* r,
+`)
+
+var pulseaudioConnectedPlugSecComp = []byte(`
 setsockopt
+getsockopt
 connect
 sendto
-`
-
-// NewPulseAudioInterface returns a new "pulseaudio" interface.
-func NewPulseAudioInterface() interfaces.Interface {
-	return &commonInterface{
-		name: "pulseaudio",
-		connectedPlugAppArmor: pulseaudioConnectedPlugAppArmor,
-		connectedPlugSecComp:  pulseaudioConnectedPlugSecComp,
-		reservedForOS:         true,
-		autoConnect:           true,
+shmctl
+getsockname
+getpeername
+sendmsg
+recvmsg
+`)
+
+var pulseaudioPermanentSlotAppArmor = []byte(`
+# When running PulseAudio in system mode it will switch to the at
+# build time configured user/group on startup.
+capability setuid,
+capability setgid,
+
+capability sys_nice,
+capability sys_resource,
+
+@{PROC}/self/exe r,
+/etc/machine-id r,
+
+# Audio related
+@{PROC}/asound/devices r,
+@{PROC}/asound/card** r,
+/dev/snd/pcm* rw,
+/dev/snd/control* rw,
+/dev/snd/timer r,
+/sys/**/sound/** r,
+
+# For udev
+network netlink raw,
+/sys/devices/virtual/dmi/id/sys_vendor r,
+/sys/devices/virtual/dmi/id/bios_vendor r,
+/run/udev/data/** r,
+
+owner /{,var/}run/pulse/ rwk,
+owner /{,var/}run/pulse/native rwk,
+owner /{,var/}run/pulse/pid rwk,
+owner /{,var/}run/pulse/.config/ rwk,
+owner /{,var/}run/pulse/.config/pulse/ rwk,
+
+# Shared memory based communication with clients
+/{run,dev}/shm/pulse-shm-* rwkcix,
+owner /{,var/}run/pulse/.config/pulse/cookie rwk,
+`)
+
+var pulseaudioPermanentSlotSecComp = []byte(`
+personality
+setpriority
+setsockopt
+getsockname
+bind
```

Comment:
All of these I imagine are for the UNIX socket, not INET, so, no, let's leave this as is.